src/common/native/main/OpenSettingsHandler.ts

```diff
@@ -0,0 +1,20 @@
+import m from "mithril"
+import { LoginController } from "../../api/main/LoginController.js"
+
+/**
+ * Handles requests for opening settings paths from native.
+ */
+export class OpenSettingsHandler {
+	constructor(private readonly logins: LoginController) {}
+
+	async openSettings(path: string): Promise<void> {
+		await this.logins.waitForFullLogin()
+
+		console.log("Trying to open settings page", path)
+		if (this.logins.isUserLoggedIn()) {
```

Comment:
you are waiting for full login (partial login is probably enough?) and then still check if you are logged in or not.

If possible I'd use mithril's path parameters handling rather than interpolating on our own, specially with not-so-trusted input.